### PR TITLE
[Gluten-4585][VL] Support spark.sql.files.ignoreMissingFiles=true

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?

Fixes: #4585 
Upstream PR: https://github.com/facebookincubator/velox/pull/8615, https://github.com/facebookincubator/velox/pull/8662

## How was this patch tested?

UT.

### CODE REVIEW

**Comment by zhli1142015** [comment]
@rui-mo , @PHILO-HE , @JkSelf , could you help to review this PR? Thanks.

**Review by PHILO-HE — COMMENTED** [review]
Looks good! Just a tiny comment for test. Thanks!

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -367,7 +367,7 @@ class VeloxTestSettings extends BackendTestSettings {
     .excludeByPrefix("SPARK-22790")
     // plan is different cause metric is different, rewrite
     .excludeByPrefix("SPARK-25237")
-    // ignoreMissingFiles mode, wait to fix
+    // ignoreMissingFiles mode, rewrite
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:370]: Could we desrible why we need rewrite these tests? Thanks.

Comment (zhli1142015) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala]: Updated, thanks

File: gluten-ut/spark33/src/test/scala/org/apache/spark/sql/GlutenFileBasedDataSourceSuite.scala [hunk 2]
```diff
@@ -174,4 +179,72 @@ class GlutenFileBasedDataSourceSuite extends FileBasedDataSourceSuite with Glute
...
+        // Rewrite this test as error msg is different from velox
+        for {
+          (ignore, options, sqlConf) <- Seq(
+            // Set via SQL Conf: leave options empty
+            ("true", Map.empty[String, String], "true"),
+            ("false", Map.empty[String, String], "false")
+          )
+          sources <- Seq("", format)
+        } {
+          if (BackendTestUtils.isVeloxBackendLoaded()) {
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply gluten-ut/spark33/src/test/scala/org/apache/spark/sql/GlutenFileBasedDataSourceSuite.scala]: Updated, thanks.

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/clickhouse/ClickHouseTestSettings.scala [hunk 3]
```diff
@@ -319,6 +319,7 @@ class ClickHouseTestSettings extends BackendTestSettings {
   enableSuite[GlutenFileBasedDataSourceSuite]
     .exclude("SPARK-23072 Write and read back unicode column names - csv")
     .excludeByPrefix("Enabling/disabling ignoreMissingFiles using")
+    .excludeByPrefix("Gluten - Enabling/disabling ignoreMissingFiles using")
```
[the remark this answers is not in the document]
Comment (PHILO-HE) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/clickhouse/ClickHouseTestSettings.scala:322]: Or use `excludeGlutenTestsByPrefix`.

**Review by PHILO-HE — DISMISSED** [review]
Looks good! Thanks!

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Pull request merged**